src/main/java/com/microsoft/lst_bench/client/JDBCConnection.java

```diff
@@ -18,28 +18,51 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** A JDBC connection. */
 public class JDBCConnection implements Connection {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(JDBCConnection.class);
+
   private final java.sql.Connection connection;
+  private final int max_num_retries;
 
-  public JDBCConnection(java.sql.Connection connection) {
+  public JDBCConnection(java.sql.Connection connection, int max_num_retries) {
     this.connection = connection;
+    this.max_num_retries = max_num_retries;
   }
 
   @Override
   public void execute(String sqlText) throws ClientException {
-    try (Statement s = connection.createStatement()) {
-      boolean hasResults = s.execute(sqlText);
-      if (hasResults) {
-        ResultSet rs = s.getResultSet();
-        while (rs.next()) {
-          // do nothing
+    Exception last_error = null;
+    int error_count = 0;
+
+    while (error_count < this.max_num_retries) {
+      try (Statement s = connection.createStatement()) {
+        boolean hasResults = s.execute(sqlText);
+        if (hasResults) {
+          ResultSet rs = s.getResultSet();
+          while (rs.next()) {
+            // do nothing
+          }
         }
+        return;
+
+      } catch (Exception e) {
+        last_error = e;
+        error_count++;
       }
-    } catch (Exception e) {
-      throw new ClientException(e);
+    }
+
+    if (last_error != null) {
+      LOGGER.warn(
```

Comment:
Could we wrap the exception into our own exception with the additional message so it is also recorded in the telemetry?